pkg/bee/chunk.go

```diff
@@ -3,90 +3,41 @@ package bee
 import (
 	"encoding/binary"
 	"fmt"
-	"hash"
 	"math/rand"
 
 	"github.com/ethersphere/bee/pkg/swarm"
-	bmtlegacy "github.com/ethersphere/bmt/legacy"
-	"golang.org/x/crypto/sha3"
 )
 
-const (
-	// MaxChunkSize represents max chunk size in bytes
-	MaxChunkSize = 4096
-	spanInfoSize = 8
-)
-
-// Chunk represents Bee chunk
-type Chunk struct {
```

Comment:
This type represents this pkg's type which has slightly different purpose and usage than the swarm.Chunk type.
In the future it can contain some additional fields needed for beekeeper.
It can be used as a wrapper around swarm.Chunk type, i.e. swarm.Chunk can be one filed in this struct.
Logic of the whole pkg is designed like that - this way underlaying technologies can be easily managed, new can be added or old ones can be even replaced.

What was the reason for replacing it with the other pkg's type?

Reply:
the main problem is that it is not a 1:1 implementation. in some details it is incorrectly enforcing some constraints such as size (maximum data part of the chunk is always 4096, span is appended to that, but in the implementation here the data of the chunk is 4096 minus span, which is incorrect), and in other cases it separates span and size, which is incorrect since they are in normal swarm chunks not separate from the data. i found it a bit confusing to maintain a parallel implementation that is no 1:1, it also prevents interoperability with helper functions and other component in `bee` (which is a dependency).

Reply:
I think that the original idea was that Chunk in bee and chunk in beekeeper have different purposes and that beekeeper requirements do not have to be implemented in bee. The correction in data size should be trivial without so much changes.

I will give my review, but this PR has both architecture refactoring and new and changed checks with a bit larger number of lines changed. It would be better to have PRs that are focused on one purpose, if possible.

Reply:
Would this be helpful?

`type Chunk struct {
	Swarm swarm.Chunk
}`

Reply:
> I think that the original idea was that Chunk in bee and chunk in beekeeper have different purposes and that beekeeper requirements do not have to be implemented in bee. The correction in data size should be trivial without so much changes.

I thought that you gave your consent to do this change, we discussed this no? Maybe I got you wrong.
OK.
You want the old chunk type back?